docs/administration/recent_activity/recent_activity.md

```diff
@@ -0,0 +1,231 @@
+
+In the following example, an event subscriber is subscribing to an event dispatched by a custom feature. This event has the information needed by a log entry (see details after the example).
+
+```php
+[[= include_file('code_samples/recent_activity/src/EventSubscriber/MyFeatureEventSubscriber.php') =]]
+```
+
+`ActivityLogService::build` function returns an `Ibexa\Contracts\ActivityLog\Values\CreateActivityLogStruct` which can then be passed to `ActivityLogService::save`.
+
+`ActivityLogService::build` has three arguments:
+
+* `$className` is the FQCN of the object actually manipulated by the feature. For example `Ibexa\Contracts\Core\Repository\Values\Content\Content::class`
+* `$id` is the ID or identifier of the manipulated object. For example, the Content ID cast to string.
+* `$action` is the identifier of the performed object manipulation. For example, `create`, `update` or `delete`.
+
+The returned `CreateActivityLogStruct` is always related to the currently logged-in user.
+
+If the object you log an activity on can become unavailable (like after a `delete` action), you might want to also log the name the object has at log time to be able to display it even when the object becomes unavailable. To add this name, use `CreateActivityLogStruct::setName` before saving the log entry.
+
+#### Context group
+
+If you log several entries at once, you can group them into a context.
+A context group counts as one item in regard to `activity_logs_limit` configuration and `ActivityLogService::findGroups`'s `$limit` argument.
+
+To open a context group, use `ActivityLogService::prepareContext` which has 2 arguments:
+
+* `$source` is describing (usually through a short identifier) what is triggering the set of actions. For example, some already existing source are `web` (like for actions from the Back Office), `graphql`, `rest` and `migration`.
+* `$description` is an optional more specific contextualisation. For example, `migration` context source is associated with the migration file name in its context description.
+
+To close a context group, use `ActivityLogService::dismissContext`.
+
+In the following example, several actions are logged into one context group, even actions triggered by cascade outside the piece of code:
+
+- `my_feature`
+    - `init`
+    - `create`
+    - `publish`
+    - `simulate`
+    - `complete`
+
+``` php
+[[= include_file('code_samples/recent_activity/src/Command/ActivityLogContextTestCommand.php', 63, 83) =]]
+```
+
+Context groups can't be nested.
+If a new context is prepared while a context is already grouping log entries, this new context will be ignored. (Nested context might be supported in the future.)
+To start a new context, make sure to previously dismiss the existing one.
+
+As the context are not really user-friendly, by default, contexts are not shown in the Back Office.
```

Comment:
I don't think that we should write "they are not really user-friendly". I'd just keep second part of the sentence.

Reply:
```suggestion
```